pkg/util/resource/monitor_test.go

```diff
@@ -0,0 +1,18 @@
+package resource
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Monitor(t *testing.T) {
+	m, err := NewMonitor(map[Type]float64{}, prometheus.DefaultRegisterer)
+
+	m.scanners[CPU] = &noopScanner{}
+	m.containerLimit[CPU] = 1
+	m.utilization[CPU] = 0.5
+
+	require.NoError(t, err)
```

Comment:
I am ok with such unit test but probably we can add an integ test with those new flags, which should be able to capture more problems I hope

Reply:
That's true, let me add an integ test at least just to spin up cortex with the resource monitor configs enabled.

Reply:
Added `integration/resource_based_limiter_test.go`